src/list.rs

```diff
@@ -0,0 +1,32 @@
+use anyhow::Result;
+
+use async_zip::{read::seek::ZipFileReader, ZipEntry};
+use cobalt_aws::s3::S3Object;
+
+use crate::aws::S3ObjectSeekableRead;
+
+/// Provide a method of iterating over the details in of archived
+/// files without reading the entire archive.
+pub struct ZipEntries<'a>(ZipFileReader<S3ObjectSeekableRead<'a>>);
+
+impl<'a> ZipEntries<'a> {
+    ///Create a new `ZipEntries` for the `src` object.
+    pub async fn new<'b>(
+        client: &'b aws_sdk_s3::Client,
+        src: &'b S3Object,
+        bytes_before_fetch: Option<u64>,
```

Comment:
I wonder whether we need to expose `bytes_before_fetch` to the caller here at all, or whether just always using the default `None` would be appropriate`. IIUC the way this gets used in practice, it will do a couple of backward seeks in order to find the start of the central directory, and then stream it forward from there, so there's not a lot of potential performance tuning the caller could do by tweaking this value.